qhub/provider/cicd/gitlab.py

```diff
@@ -53,13 +54,19 @@ def gen_gitlab_ci(config):
         f"git checkout {branch}",
         f"{pip_install}",
         "qhub deploy --config qhub-config.yaml --disable-prompt --skip-remote-state-provision",
+    ]
+
+    commit_render_script = [
         "git config user.email 'qhub@quansight.com'",
         "git config user.name 'gitlab ci'",
         "git add .",
         "git diff --quiet && git diff --staged --quiet || (git commit -m '${COMMIT_MSG}'",
         f"git push origin {branch})",
     ]
 
+    if os.environ.get("QHUB_PREVENT_COMMIT_RENDER", "no") != "yes":
```

Comment:
Personally I'd prefer if this was an option in the qhub-config.yaml file. @viniciusdc and @iameskild what are your thoughts? My reasoning for this is so that there is one and only one way to do things. I do agree that we should have a way to set config from environment variables though. E.g. `cicd.commit_render: true` could be represented via an environment variable like `QHUB__cicd__commit_render=true` or something along those lines.